scripts/set-ports.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash -eu
```

Comment:
This is a very cool script and I like what it does.  But I'm not sure it's the behavior we want.  I don't think we have a use case where we need to blindly install servers without first knowing what ports are free.  Whoever is installing this should be able to set the ports they want, and then have it either succeed or fail.  If it succeeds, then they know they can connect to the server and DB on the ports they entered.  If it fails, they know they need to try again with different ports.  Although the below is cool, it's less ideal for ports to be selected automatically from what is free.  I'd rather it fail hard and notify that there is an issue.